### DISCUSSION TITLE
docker not found / exit code 127

### DISCUSSION THREAD

**Discussion by DanielRuf** [community/community#25409] [Actions]
<p>It seems there is some issue when we use the node:latest image.</p>
<p><a href="https://github.com/bitExpert/mattermost-client-node/commit/a4789d1163b65ed8c11c3de5a9866c95816681c2/checks?check_suite_id=413831133" rel="nofollow noopener">https://github.com/bitExpert/mattermost-client-node/commit/a4789d1163b65ed8c11c3de5a9866c95816681c2/checks?check_suite_id=413831133</a></p>
<p>Not sure why.</p>
<p>##[error]Process completed with exit code 127.</p>
<p>Run docker pull mattermost/mattermost-preview &amp;&amp; \</p>
<p>docker pull mattermost/mattermost-preview &amp;&amp; \</p>
<p>docker run --name mattermost-preview -d --publish 8065:8065 --add-host dockerhost:127.0.0.1 mattermost/mattermost-preview</p>
<p>shell: sh -e {0}</p>
<p>/__w/_temp/ad18eb6b-44b4-4e13-a5e2-a033258c5f34.sh: 1: /__w/_temp/ad18eb6b-44b4-4e13-a5e2-a033258c5f34.sh: docker: not found</p>
<p>##[error]Process completed with exit code 127.</p>

**Comment by Yanjingzhu** [comment]
<p>I checked your workflow file. Job <strong>Node latest</strong> runs on container with image node:latest. </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/6/60b63cd7e98d11851cfbbb8916ab0fbc769b4aac.png" data-download-href="/uploads/short-url/dNyhkFdZQ3GNlGbVVe71hBpjZHS.png?dl=1" title="container job.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181090991-7e52c230-93e9-4a91-a393-93445d01557e.png" alt="container job.png" data-base62-sha1="dNyhkFdZQ3GNlGbVVe71hBpjZHS" width="690" height="162" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/6/60b63cd7e98d11851cfbbb8916ab0fbc769b4aac_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">container job.png</span><span class="informations">905×213 5.84 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>All steps that don’t already specify a container will run in node latest container. So the docker commands will run on node latest container. You got docker: not found error because docker is not installed on node latest container. </p>
<p>  There is the document for  <strong>jobs.&lt;job_id&gt;.container:</strong> <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idcontainer" rel="nofollow noopener">https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idcontainer </a></p>

**Comment by DanielRuf** [comment]
<p>Right. But how would you create two different steps in one job which run on two different platforms? It’s probably not possible. At least the GHA docs are not very helpful here.</p>
<p>My main problem is the lack of documentation and knowledge, see also <a href="https://github.com/actions/setup-node/issues/61" rel="nofollow noopener">https://github.com/actions/setup-node/issues/61</a></p>

**Comment by DanielRuf** [comment, marked as answer]
<p>Ok so it seems ‘*’ works for the setup-node action but is not tested / documented as it seems.</p>

**Comment by Jmunsey859** [comment]
<p>apt-get install docker* -this solved the problem for me in kali-linux</p>

**Comment by DanielRuf** [comment]
<p>Not really. It depends on your project setup.</p>
